src/librbd/api/Trash.cc

```diff
@@ -145,47 +138,80 @@ int Trash<I>::move(librados::IoCtx &io_ctx, rbd_trash_image_source_t source,
     return -EOPNOTSUPP;
   }
 
-  image_id = ictx->id;
-  ictx->owner_lock.get_read();
-  if (ictx->exclusive_lock != nullptr) {
-    ictx->exclusive_lock->block_requests(0);
+  if (image_id.empty()) {
```

Comment:
You still need `image_id = ictx->id` (removed above). For the case when `move_by_id` is called with empty `image_id` but non empty `image_name`.

Reply:
oops, yeah. I'm surprised there's no test failure..